File: FinTech/FinTech.API/Models/Transaction.cs
using FinTech.API.Enum;

namespace FinTech.API.Models
{
    public class Transaction
    {
        public Guid Id { get; set; }

        public string IdempotencyKey { get; set; } = string.Empty;

        public TransactionType Type { get; set; }

        public decimal Amount { get; set; }

        public TransactionStatus Status { get; set; }

        public Guid? LoanId { get; set; }

        public string Description { get; set; } = string.Empty;

        public DateTime CreatedAt { get; set; }
    }
}

File: fintech-frontend/types/transaction.ts
export interface Transaction {
  id: string;
  idempotencyKey: string;
  type: TransactionType;
  amount: number;
  loanId: string;
  description: string;
  createdAt: string;
}

export enum TransactionType {
  PAYMENT = 0,
  DISBURSEMENT = 1,
}

export const TransactionTypeLabel: Record<
  TransactionType,
  string
> = {
  [TransactionType.PAYMENT]:
    "Payment",

  [TransactionType.DISBURSEMENT]:
    "Disbursement",
};

export enum TransactionStatus {
  PENDING = 0,
  COMPLETED = 1,
  FAILED = 2,
}

export const TransactionStatusLabel:
Record<TransactionStatus, string> = {
  [TransactionStatus.PENDING]:
    "Pending",

  [TransactionStatus.COMPLETED]:
    "Completed",

  [TransactionStatus.FAILED]:
    "Failed",
};